hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsClient.java

```diff
@@ -217,8 +220,19 @@ protected AbfsPerfTracker getAbfsPerfTracker() {
     return abfsPerfTracker;
   }
 
-  ExponentialRetryPolicy getRetryPolicy() {
-    return retryPolicy;
+  ExponentialRetryPolicy getExponentialRetryPolicy() {
+    return exponentialRetryPolicy;
+  }
+
+  LinearRetryPolicy getLinearRetryPolicy() {
+    return linearRetryPolicy;
+  }
+
+  public RetryPolicy getRetryPolicy(final String failureReason) {
+    if (failureReason == null || !failureReason.equals(CONNECTION_TIMEOUT_ABBREVIATION))
```

Comment:
we can have `!CONNECTION_TIMEOUT_ABBREVIATION.equals(failureReason)`. would not need null-check then.

Reply:
Nice!!
Taken